src/App.php

```diff
@@ -468,7 +469,7 @@ public function initIncludes()
         $this->requireJs($this->cdn['serialize-object'] . '/jquery.serialize-object.min.js');
 
         // Agile UI
-        $this->requireJs($this->cdn['atk'] . '/atkjs-ui.min.js');
+        $this->requireJs($this->cdn['atk'] . ($this->useVueJs ? '/atkjs-vue.min.js' : '/atkjs-ui.min.js'));
```

Comment:
not ideal, we should always load atk and optionally the 2nd js for vue, otherwise use might need to download more data than now (think like login without vue and app with vue)

if not possible, then maybe better even not to split... :(

Reply:
Yes, can split, just thought it would be better to only have one request instead of two if Vue is needed.

Reply:
requests are not an issue, like css loads several fonts etc, 2, 3, requests for JS instead 1 is absolutelly fine

Reply:
Will load js per component no need for two separate bundles.

Reply:
If I will need 20 components (I know, there is currently not that many), what will be the total size?

Reply:
there will be at least one RTT with this approach if we render new/unseen component often

Reply:
> If I will need 20 components (I know, there is currently not that many), what will be the total size and what is the avg. size per component?

This really depend on the component itself. For example, tree-item-selector size is only 10kb while query-builder does 400k. I also intend to use more of external Vue component, like v-calendar, which is bigger in size. it only make sense to load it when needed. Otherwise, js bundle keep increase in size while we are using only 10% of it most of the time.

> there will be at least one mroe RTT with this approach if we render new/unseen component often

Not sure what you mean here...

Reply:
RTT - round time trip - basically one latency, which can be 100 - 500 ms if you are really far away from the server, but it should be only once per browser session if cached, so ok probably

10kb - good, but what will happen if there will be multiple simillar components? If there will be some dep with size like 500kb, I think, it will be packed everywhere... So I think sooner or later we will need more lib than component oriented approach.

Reply:
> 10kb - good, but what will happen if there will be multiple simillar components? If there will be some dep with size like 500kb, I think, it will be packed everywhere... So I think sooner or later we will need more lib than component oriented approach.

I think this is the right approach for now. When we create a Vue 'app' instance, we need to supply our root component and this is what this does. Then from the root, you can define/add other components that your need for your app instance.

Reply:
I have nothing to merge this, seems well made (not too much added code :) )